### ISSUE TITLE
concurrent running processes on render

### ISSUE THREAD

**Issue by ryandotsmith** [indexsupply/x#151]
I've noticed that E2PG reports database duplication errors on deploy and restart.

```
writing indexed data: ERROR: duplicate key value violates unique constraint "erc20_transfers_unique" (SQLSTATE 23505)
```

I realized that render is running 2 copies of E2PG for short periods of time during deploy / restart. Aside from the PG duplication error messages, E2PG is resilient to this behavior. What happens is: 1 of the E2PG processes will process a batch of blocks. A single E2PG process will succeed while the others encounter a duplication error and then rollback their postgres transactions on crash.

It's possible to detect this scenario to provide an error message to indicate what is going on. To do this, an E2PG task could acquire a pg_advisory lock before starting to work on a task. If it cannot acquire the lock then it can print a useful message and exit.

Such a mechanism could provide redundancy or parallelization of task processing in the future.

TL;DR not a major issue

**Comment by ryandotsmith** [comment]
I found a better solution: https://github.com/indexsupply/x/commit/1953958d4cdd0043566d4fb0a484ead7c14cb5a4
